tools/content-generator/src/main/java/org/projectnessie/tools/contentgenerator/cli/DeleteContent.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (C) 2022 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.tools.contentgenerator.cli;
+
+import java.util.List;
+import org.projectnessie.client.api.NessieApiV1;
+import org.projectnessie.error.NessieConflictException;
+import org.projectnessie.error.NessieNotFoundException;
+import org.projectnessie.model.Branch;
+import org.projectnessie.model.CommitMeta;
+import org.projectnessie.model.ContentKey;
+import org.projectnessie.model.Operation;
+import org.projectnessie.model.Reference;
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Model.CommandSpec;
+import picocli.CommandLine.Option;
+import picocli.CommandLine.Spec;
+
+/** Deletes content objects. */
+@Command(name = "delete", mixinStandardHelpOptions = true, description = "Delete content objects")
+public class DeleteContent extends AbstractCommand {
+
+  @Option(
+      names = {"-r", "--branch"},
+      description = "Name of the branch where to make changes, defaults to 'main'")
+  private String ref = "main";
+
+  @Option(
+      names = {"-k", "--key"},
+      description = "Content key to delete",
+      required = true)
+  private List<String> key;
+
+  @Option(
+      names = {"-m", "--message"},
+      description = "Commit message")
+  private String message;
+
+  @Spec private CommandSpec spec;
+
+  @Override
+  public void execute() throws NessieNotFoundException, NessieConflictException {
+    try (NessieApiV1 api = createNessieApiInstance()) {
+      ContentKey contentKey = ContentKey.of(key);
+
+      Reference refInfo = api.getReference().refName(ref).get();
+      if (refInfo.getType() != Reference.ReferenceType.BRANCH) {
```

Comment:
Think you can skip this check - Nessie server will complain ;)

Reply:
fixed